### PR TITLE
For #10342: Populate top sites with frequently opened sites

### LINKED ISSUE #10342: FNX-3447 ⁃ Populate top sites with frequently opened sites
### User Story
As a user, I want to see my frequently accessed sites on the homescreen in a useful way, so I can reference them to complete my routine tasks faster. 

### Dependencies
* [ ] https://github.com/mozilla-mobile/android-components/issues/7261
* [x] https://github.com/mozilla/application-services/issues/2163
* [x] https://github.com/mozilla-mobile/android-components/issues/7094

### Acceptance Criteria
* I can disable this feature
* If enabled, I see my frequently accessed sites on my homescreen
* There is a cap on # of automatically generated top sites to avoid cluttering the homescreen
* Users can turn off this feature in settings (just the auto-populated top sites, not the user added ones) (This is a requirement for shipping this feature) 
* I can customize how my top sites show up on the homescreen (e.g. choose how many sites or rows to see)

Nice to have:

* Ability to rename top sites #9548
* Rearrange Top Sites items #8305

### PR DESCRIPTION
**Pull request by gabrielluong**
Fixes #10342

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### After merge
- [ ] **Milestone**: Make sure issues finished by this pull request are added to the [milestone](https://github.com/mozilla-mobile/fenix/milestones) of the version currently in development.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -522,6 +522,8 @@
     <string name="collection_open_tabs">Open tabs</string>
     <!-- Text for the menu button to remove a top site -->
     <string name="remove_top_site">Remove</string>
+    <!-- Text for the menu button to delete a top site from history -->
+    <string name="delete_from_history">Delete from History</string>
```
Comment (ekager) [inline app/src/main/res/values/strings.xml:526]: Should `History` be capitalized here?

Comment (gabrielluong) [reply app/src/main/res/values/strings.xml]: Checked with Vesta and renamed it to "Delete from history". Thank you for the nudge!

**Comment by jonalmeida** [comment]
@gabrielluong @ekager we're going to merge this PR with the other breaking changes from https://github.com/mozilla-mobile/fenix/pull/13933

If we do them separately, it'll take much longer and we want all of these changes together either way.

**Comment by ekager** [comment]
Sounds good!

**Comment by csadilek** [comment]
Closing for: https://github.com/mozilla-mobile/fenix/pull/14241

**Pull request closed without merging**